src/gt4py/next/iterator/transforms/infer_domain.py

```diff
@@ -157,23 +160,23 @@ def infer_as_fieldop(
             raise ValueError(f"Unsupported expression of type '{type(in_field)}'.")
         input_ids.append(id_)
 
-    accessed_domains: ACCESSED_DOMAINS = _extract_accessed_domains(
+    inputs_accessed_domains: ACCESSED_DOMAINS = _extract_accessed_domains(
```

Comment:
Reasoning for the following changes in this function: This dict contains as keys not only the symref inputs, but also temporary ids. The symrefs are already added to the result dict by the loop below, while the temporary ids should not be in the result anyway. as such do not use this dict as the starting point for the domain union in the loop below.